src/index.ts

```diff
@@ -55,34 +55,56 @@ export async function run(): Promise<void> {
     const newImage = getFullImageName(image, tagsList[0]);
     const useOCI = core.getInput(Inputs.OCI) === "true";
 
-    const arch = getArch();
-    const platform = getPlatform();
+    const archs = getArch();
+    const platforms = getPlatform();
 
-    if (arch && platform) {
+    // core.debug(`Archs ---> ${archs.toString()}`);
+    // core.debug(`Platforms ---> ${platforms.toString()}`);
+
+    if ((archs.length > 0) && (platforms.length > 0)) {
         throw new Error("The --platform option may not be used in combination with the --arch option.");
     }
 
     if (containerFiles.length !== 0) {
-        await doBuildUsingContainerFiles(cli, newImage, workspace, containerFiles, useOCI, arch, platform, labelsList);
+        await doBuildUsingContainerFiles(cli, newImage, workspace, containerFiles, useOCI,
+            archs, platforms, labelsList);
     }
     else {
-        if (platform) {
+        if (platforms.length > 0) {
             throw new Error("The --platform option is not supported for builds without containerfiles.");
         }
-        await doBuildFromScratch(cli, newImage, useOCI, arch, labelsList);
+        await doBuildFromScratch(cli, newImage, useOCI, archs, labelsList);
     }
 
-    if (tagsList.length > 1) {
+    if ((archs.length > 0) || (platforms.length > 0)) {
+        core.info(`Creating manifest with tag${tagsList.length !== 1 ? "s" : ""} "${tagsList.join(", ")}"`);
+        for (const tag of tagsList) {
+            const manifestName = getFullImageName(image, tag);
+            await cli.manifestCreate(manifestName);
+
+            for (const arch of archs) {
+                const tagSuffix = getTagSuffix(arch);
+                await cli.manifestAdd(manifestName, `${newImage}-${tagSuffix}`);
+            }
+
+            for (const platform of platforms) {
+                const tagSuffix = getTagSuffix(platform);
+                await cli.manifestAdd(manifestName, `${newImage}-${tagSuffix}`);
+            }
+        }
+    }
+    else if (tagsList.length > 1) {
         await cli.tag(image, tagsList);
     }
+
```

Comment:
at the bottom could we add a 'summary' message. like 'Successfully built <image, tags list>' or 'Successfully built images <image, tags list> and manifest <manifest>`